omod/src/main/webapp/resources/scripts/jquery-ui/js/openmrsSearch.js

```diff
@@ -1075,14 +1075,16 @@ function OpenmrsSearch(div, showIncludeVoided, searchHandler, selectionHandler,
             return Math.ceil(this._table.fnSettings()._iDisplayStart / this._table.fnSettings()._iDisplayLength) + 1;
         },
         _updatePageInfo: function(searchText) {
-            textToDisplay = omsgs.viewingResultsFor.replace("_SEARCH_TEXT_", "'<b>"+searchText+"</b>'");
+            if ((/^[a-zA-Z0-9_\-]+$/.test(searchText))) {
```

Comment:
Isn't it enough to sanitizeHtml? Why this "if"?

Reply:
@rkorytkowski It wasn't when I tested. I still found the same issue. Secondly Its better handled at the generation point for the HTML than at the use level of the OpenmrsSearch function.